### PR TITLE
BREAKING CHANGE: Simplify bem block API. Improve package bundle 

### PR DESCRIPTION
**Pull request by slavik-chapelskyi**
- Add size limit
- Replace babel & build with esbuild
- Run test with ts-jest (out of the box type checks)

BREAKING CHANGE:
Remove `b` & `blazeHelper`. Use the return function of `createBlock` instead.
The definition has not changed, only how its get called

### Usage
```
import {b, createBlock} from '@shelf/bem'
// before
const block = createBlock('heading') 
b(block)

// after 
import {createBlock} from '@shelf/bem'
const b = createBlock('heading') 
b()
```
All changes
| Before| After | 
|--------|--------|
| `b(block)`  | `b()` |
|	`b(block, 'list')` | b('list') |
| `b(block, 'list-item', 'active')` | `b('list-item', {active: true})` | 
| `b(block, '', 'active')` | `b({active: true})` | 
| `b(block, '', ['active', 'hovered'])` | `b({active: true, hovered: true})` | 
| `b(block, 'list-item', [isActive && 'active',  isHovered && 'hovered'])` | `b('list-item', {active: isActive, hovered: isHovered})` |

### CODE REVIEW

**Review by vladholubiev — APPROVED** [review]

**Review by kchlonShelf — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -8,35 +8,29 @@
     "email": "vlad@shelf.io",
     "url": "https://shelf.io"
   },
-  "main": "lib",
-  "module": "lib/es/index.js",
+  "exports": "./lib/index.es.js",
```
Comment (Rafler) [inline package.json:11]: exports should specify separately cjs and esm

**Review by Rafler — COMMENTED** [review]
Generally, it looks good, but I think we need to make it backward compatible as it will affect styles across all apps and could cause too many breaking changes and production issues. I would suggest marking the current API as deprecated and add new export with the suggested functionality

**Review by irenhh — APPROVED** [review]

**Comment by slavik-chapelskyi** [comment]
> Generally, it looks good, but I think we need to make it backward compatible as it will affect styles across all apps and could cause too many breaking changes and production issues. I would suggest marking the current API as deprecated and add new export with the suggested functionality

With deprecating approach, I`m just concerned, we will update the package and continue using old API syntax.

File: package.json [hunk 2]
```diff
@@ -47,32 +45,38 @@
         "lines": 100,
         "statements": 100
       }
-    }
+    },
+    "preset": "ts-jest"
```
Comment (Rafler) [inline package.json:49]: maybe will be better to use @swc/jest ?

File: esbuild.mjs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import {common} from '@shelf/esbuild-config';
+
+await Promise.all(
+  ['esm', 'cjs'].map(
+    async format =>
+      await common({
+        entryPoints: ['src/index.ts'],
+        config: {
+          splitting: false,
```
Comment (Rafler) [inline esbuild.mjs:9]: is there a reason why we passing **false** to `splitting`

File: package.json (hunk 2 above)
Comment (slavik-chapelskyi) [reply package.json:49]: The real benefit of using ts-jest is that it do typechecking in test. 
For projects I would suggest speed is more important, but for package with 10 test it won't have such an effect, but with ts-jest we get extra coverage

File: esbuild.mjs (hunk 3 above)
Comment (slavik-chapelskyi) [reply esbuild.mjs:9]: CleanShot 2023-07-06 at 14 24 41@2x
It don't work in cjs

**Review by Rafler — APPROVED** [review]

**Pull request merged**